### PR TITLE
refactor: validate test case indices

### PR DESCRIPTION
**Pull request by ouuan**
## Description

Validate test case indices.

## Motivation and Context

Make it safer.

## How Has This Been Tested?

On Arch Linux.

## Checklist

- [x] If the key of a setting is changed, the `old` attribute is updated or it is resolved in SettingsUpdater.
- [x] If there are changes of the text displayed in the UI, they are wrapped in `tr()` or `QCoreApplication::translate()`.
- [x] If needed, I have opened a pull request or an issue to update the [documentation](https://github.com/cpeditor/cpeditor.github.io).
- [x] If these changes are notable, they are documented in [CHANGELOG.md](https://github.com/cpeditor/cpeditor/blob/master/CHANGELOG.md).

### CODE REVIEW

File: src/mainwindow.cpp [hunk 1]
```diff
@@ -201,6 +201,12 @@ void MainWindow::run()
 
 void MainWindow::run(int index)
 {
+    if (index < 0 || index >= testcases->count())
+    {
+        LOG_WARN(INFO_OF(index) << INFO_OF(testcases->count()));
+        return;
+    }
+
```
Comment (coder3101) [inline src/mainwindow.cpp:209]: When can indices be out of range?

Comment (ouuan) [reply src/mainwindow.cpp:209]: When we make mistakes. It shouldn't be out of range.

**Comment by ouuan** [comment]
We'd better use `LOG_DEV` from #663 :thinking:

File: src/mainwindow.cpp (hunk 1 above)
Comment (coder3101) [reply src/mainwindow.cpp:209]: Okay, so it's for finding issues quickly.

**Review by coder3101 — APPROVED** [review]

**Review by coder3101 — APPROVED** [review]

**Pull request merged**